packages/core/src/model/types.ts

```diff
@@ -1 +1,15 @@
+/* eslint-disable */
+import { Contract, ContractTransaction } from "ethers"
+
 export type Falsy = false | 0 | '' | null | undefined
+
+export type TypedContract = Contract & {
+    functions: Record<string, (...args: any[]) => any>
+}
+
+export type ContractFunctionNames<T extends TypedContract> = keyof { [P in keyof T['functions'] as ReturnType<T['functions'][P]> extends Promise<ContractTransaction> ? P : never] : void } & string
```

Comment:
eslint for some reason finds a syntax error even though there is none. Probably something with version mismatch. Will have to check in future PR